services/brig/test/integration/API/Team/Util.hs

```diff
@@ -47,22 +57,53 @@ createUserWithTeam brig galley = do
             , "password"        .= defPassword
             , "team"            .= newTeam
             ]
-    bdy <- decodeBody <$> post (brig . path "/i/users" . contentJson . body p)
-    let (Just uid, Just (Just tid)) = (userId <$> bdy, userTeam <$> bdy)
-    (team:_) <- view Team.teamListTeams <$> getTeams uid galley
+    Just user <- decodeBody <$> post (brig . path "/i/users" . contentJson . body p)
+    let Just tid = userTeam user
+    (team:_) <- view Team.teamListTeams <$> getTeams (userId user) galley
     liftIO $ assertBool "Team ID in registration and team table do not match" (tid == view Team.teamId team)
-    selfTeam <- userTeam . selfUser <$> getSelfProfile brig uid
+    selfTeam <- userTeam . selfUser <$> getSelfProfile brig (userId user)
     liftIO $ assertBool "Team ID in self profile and team table do not match" (selfTeam == Just tid)
-    return (uid, tid)
+    return (userId user, tid)
 
-addTeamMember :: Galley -> TeamId -> Team.NewTeamMember -> Http ()
```

Comment:
This function went away because it always led to an inconsistency: Galley thought the user was in the team, but Brig thought it wasn't.